homeassistant/components/vallox/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow for the Vallox integration."""
+from __future__ import annotations
+
+from contextlib import suppress
+import ipaddress
+import logging
+import re
+from typing import Any
+
+from vallox_websocket_api import Vallox
+from vallox_websocket_api.exceptions import ValloxApiException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+    }
+)
+
+
+def host_valid(host: str) -> bool:
+    """Return True if hostname or IP address is valid."""
+    with suppress(ValueError):
+        if ipaddress.ip_address(host).version in [4, 6]:
+            return True
+
+    disallowed = re.compile(r"[^a-zA-Z\d\-]")
+    return all(x and not disallowed.search(x) for x in host.split("."))
+
+
+async def validate_host(hass: HomeAssistant, host: str, name: str) -> dict[str, Any]:
+    """Validate that the user input allows us to connect."""
+
+    if not host_valid(host):
+        raise InvalidHost(f"Invalid host: {host}")
+
+    client = Vallox(host)
+
+    try:
+        info = await client.get_info()
+    except (OSError, ValloxApiException) as err:
+        raise CannotConnect("Failed to connect") from err
+
+    return {"title": name, "model": info["model"]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the Vallox integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, data: dict[str, Any]) -> FlowResult:
+        """Handle import from YAML."""
+        name = data.get(CONF_NAME, DEFAULT_NAME)
```

Comment:
Usually we don't ask for a name since the user can rename this later.

Reply:
We might be stuck keeping it in this case since yaml allowed it

Reply:
If the info returned by `client.get_info()` has a name, we can drop `name` here and just use that instead

Reply:
> If the info returned by `client.get_info()` has a name, we can drop `name` here and just use that instead which means less manual data entry for the user

I need to check what's available, but for new ones added with the config flow that would work if it's available, but can we do that for import of existing YAML configuration? Not sure I grasp all the consequences of doing it, so happy to be proven wrong.

Reply:
The name can affect how the entity id is created so we probably need to keep it for imports, but for new setups, ideally we get it from the client if it has it

Reply:
The `get_info()` method returns model (which could be usable as a name 🤷), SW version and some kind of UUID.

One caveat though, according to the Vallox library, the model can be `Unknown` in some cases, so we might need to handle that case and set it to something more sane. Or what do you think?

Reply:
f

Reply:
The default should be fine here.  It can always be adjusted later if we find its an issue